### DISCUSSION TITLE
How to enable new pprof labels features?

### DISCUSSION THREAD

**Discussion by mgabeler-lee-6rs** [golang/vscode-go#3173] [Help] [answered]
The [v0.41.0 release notes](https://github.com/golang/vscode-go/releases/tag/v0.41.0) mention a feature I'm very interested in: labeling goroutines.

I've updated delve & the vscode extension:
```
$ dlv version
Delve Debugger
Version: 1.22.0
Build: $Id: 61ecdbbe1b574f0dd7d7bad8b6a5d564cce981e9 $
$ code --list-extensions --show-versions | grep go
golang.go@0.41.0
```

And updated my code to add labels to my goroutines, but nothing is showing in the Call Stack view for them. The delve code / PR implies that the DAP client needs to explicitly tell it which ones to show: https://github.com/go-delve/delve/blob/f32818c9e36e6c6e84a07131c08f994ee38fa156/service/dap/command.go#L67-L72

However, unlike, say, `hideSystemGoroutines`, I can't find anything in the vscode-go extension that would set this flag.

Am I missing something, or are the release notes misleading here and this feature isn't actually ready for use in the extension yet?

**Comment by stefanhaller** [comment, marked as answer]
You can add a `showPprofLabels` key to your launch configuration, and it does work. It will look like it doesn't in the editor, but it does.

<img width="517" alt="image" src="https://github.com/golang/vscode-go/assets/1225667/a7c50c34-0899-4b4d-abc9-6d72d21cc0b8">

It looks like we need a commit similar to 44e2d4d521 to make it known to the editor. @suzmue, could you make that change? I'm not familiar enough with this project to feel comfortable contributing a PR.

**Reply by mgabeler-lee-6rs** [reply]
Thank you, this works! It would be awesome if this can be a delve setting like the other one you reference, instead of needing to be added to every launch config.

FWIW, it wasn't even clear to me from the docs that I could put delve settings in the launch config at all.

**Comment by gm0stache** [comment]
opened a feature request to add the `pprof` labels clearly visible in the UI of the _call stack_ section headers of the debugger.
give it a vote/thumbs up if you would also like to have this: https://github.com/golang/vscode-go/issues/3247
